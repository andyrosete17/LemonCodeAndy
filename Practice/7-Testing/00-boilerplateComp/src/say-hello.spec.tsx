import React from 'react';
import { render, screen } from '@testing-library/react';
import { SayHello } from './say-hello';

describe('SayHello component', () => {
  it('should display the person name', () => {
    // Arrange
    const person = 'Jhon';

    // Act
    render(<SayHello person={person} />);

    // Assert
    const element = screen.getByRole('heading', { level: 1 });
    expect(element).not.toBeNull();
    expect(element.tagName).toEqual('H1');
  });

  it('should display the person name using snapshot testing', () => {
    // Arrange
    const person = 'Jhon';

    // Act
    const { asFragment } = render(<SayHello person={person} />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

//   it('should display the person name using snapshot testing', () => {
//     // Arrange
//     const person = 'Jhon';

//     // Act
//     const { asFragment } = render(<SayHello person={person} />);

//     // Assert
//     expect(asFragment()).toMatchInlineSnapshot(`
// <DocumentFragment>
//   <h1>
//     Hello
//     <strong>
//       Jhon
//     </strong>
//   </h1>
// </DocumentFragment>
// `);
//   });

  it('should display the person name using jest-dom', () => {
    // Arrange
    const person = 'John';

    // Act
    render(<SayHello person={person} />);

    // Assert
    const element = screen.getByRole('heading', { name: 'Hello John' });
    expect(element).toBeInTheDocument();
  });
});
