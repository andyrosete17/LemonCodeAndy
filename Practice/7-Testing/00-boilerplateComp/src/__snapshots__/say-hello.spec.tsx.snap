// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SayHello component should display the person name using snapshot testing 1`] = `
<DocumentFragment>
  <h1>
    Hello 
    <strong>
      Jhon
    </strong>
  </h1>
</DocumentFragment>
`;
